### PR TITLE
SPI Flash support, VirtualFS

### PR DESCRIPTION
**Pull request by gagarinlg**
This PR adds support for SPI flash chips on some radios.
Also LittleFS is added as a NOR flash compatible FS and a virtual file system layer is introduced as an abstraction layer between directory and file access and the different APIs of different file systems.

### CODE REVIEW

**Comment by rotorman** [comment]
Please remove .gitmodules from commit - it currently would remove LittleFS submodule.

File: radio/src/VirtualFS.h [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    #define OTX_FOURCC 0x3678746F // otx for Taranis X7 / X7S / X7 Express / X7S Express
+  #elif defined(PCBX9LITES)
+    #define OTX_FOURCC 0x3E78746F // otx for Taranis X9-Lite S
+  #elif defined(PCBX9LITE)
+    #define OTX_FOURCC 0x3C78746F // otx for Taranis X9-Lite
+  #elif defined(PCBX9D) || defined(PCBX9DP)
+    #define OTX_FOURCC 0x3378746F // otx for Taranis X9D
+  #elif defined(PCBNV14)
+    #define OTX_FOURCC 0x3A78746F // otx for NV14
+  #elif defined(PCBSKY9X)
```
Comment (rotorman) [inline radio/src/VirtualFS.h:289]: PL18 0x4878746F
https://github.com/EdgeTX/edgetx/blob/92dac552f4de61441c2279063eed17bda8f64f68/radio/src/sdcard.h#L173-L174

Comment (gagarinlg) [reply radio/src/VirtualFS.h:289]: The defined are just left over from using sdcard.h as a base.
The path definitions at the beginning of the file will stay, I thjink, an be removed from sdcard.h. But then can only happen when everysthing is moved over to the VirtualFS.

**Comment by gagarinlg** [comment]
I fixed the littlefs submodule problems and added the PL18 changes.

**Comment by rotorman** [comment]
I also get a new warning about assert, when building under MSYS2, as this is apparently defined in LittleFS:
```
radio/src/opentx.h:351: warning: "assert" redefined
  351 |   #define assert(x)
      |
In file included from radio/src/thirdparty/littlefs_v2.4.1/lfs_util.h:32,
                 from radio/src/thirdparty/littlefs_v2.4.1/lfs.h:12,
                 from radio/src/VirtualFS.h:26,
                 from radio/src/gui/colorlcd/radio_flashmanager.cpp:22:
c:\msys64\mingw64\arm-none-eabi\include\assert.h:16: note: this is the location of the previous definition
   16 | # define assert(__e) ((__e) ? (void)0 : __assert_func (__FILE__, __LINE__, \
      |
```

File: radio/src/VirtualFS.h (hunk 1 above)
Comment (raphaelcoeffic) [reply radio/src/VirtualFS.h:409]: Do we actually still need these defs now?

Comment (gagarinlg) [reply radio/src/VirtualFS.h:409]: I think so, but not here, but only in sdcard.h

Comment (raphaelcoeffic) [reply radio/src/VirtualFS.h:409]: I believe this is used only for conversions from binary format.

Comment (gagarinlg) [reply radio/src/VirtualFS.h:409]: What is the plan regarding binary format? Require 2.6 as a conversion point for future users or do you plan to keep compatibility in 2.7 and later?

Comment (raphaelcoeffic) [reply radio/src/VirtualFS.h:409]: Either 2.7 or 2.8 will probably drop binary conversions. We shall see what the community thinks.

**Pull request merged**